lizard_auth_server/tests/test_views.py

```diff
@@ -1,24 +1,215 @@
+# -*- coding: utf-8 -*-
+
+from datetime import datetime
+import json
+
+from django.contrib.auth.models import AnonymousUser
 from django.contrib.auth.models import User
 from django.core.urlresolvers import reverse
 from django.test import Client
 from django.test import TestCase
+from django.test.client import RequestFactory
+from faker import Faker
+from jwt.exceptions import ExpiredSignatureError
+from nose.tools import raises
+import jwt
+
+from lizard_auth_server.conf import settings
+from lizard_auth_server.models import gen_key
 from lizard_auth_server.tests import factories
+from lizard_auth_server.views import JWTView
+
+JWT_ALGORITHM = settings.LIZARD_AUTH_SERVER_JWT_ALGORITHM
+JWT_EXPIRATION_DELTA = settings.LIZARD_AUTH_SERVER_JWT_EXPIRATION_DELTA
+
+fake = Faker()
 
 
 class ProfileViewTestCase(TestCase):
 
     def test_smoke_as_regular_user(self):
-        self.user = User.objects.create_user('someone', 'a@a.nl', 'pass')
-        self.user_profile = factories.UserProfileF(user=self.user)
-        self.client = Client()
-        self.client.login(username='someone', password='pass')
-        result = self.client.get(reverse('index'))
+        User.objects.create_user('someone', 'a@a.nl', 'pass')
+        client = Client()
+        client.login(username='someone', password='pass')
+        result = client.get(reverse('index'))
         self.assertEquals(result.status_code, 200)
 
     def test_smoke_as_admin(self):
-        self.user = User.objects.create_superuser('admin', 'a@a.nl', 'pass')
-        self.user_profile = factories.UserProfileF(user=self.user)
-        self.client = Client()
-        self.client.login(username='admin', password='pass')
-        result = self.client.get(reverse('index'))
+        User.objects.create_superuser('admin', 'a@a.nl', 'pass')
+        client = Client()
+        client.login(username='admin', password='pass')
+        result = client.get(reverse('index'))
         self.assertEquals(result.status_code, 200)
+
+
+class JWTViewTestCase(TestCase):
+
+    def setUp(self):
+        self.factory = RequestFactory()
+        self.portal = factories.PortalF()
+        self.user = factories.UserF()
+
+    def test_absolute_url(self):
+        self.assertTrue(JWTView.is_url(fake.url()))
+
+    def test_invalid_portal(self):
+        sso_key = gen_key('Portal', 'sso_key')
```

Comment:
Die `gen_key()` ziet eruit alsof je een echte key genereert. Is het niet duidelijker qua test om te zeggen:

    sso_key = 'ha, ha, dat dacht je, hè?'